display_list/display_list_path_effect.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/display_list/display_list_path_effect.h"
+
+#include <memory>
+#include <utility>
+
+#include "include/core/SkRefCnt.h"
+#include "include/core/SkScalar.h"
+
+namespace flutter {
+
+static void DlPathEffectDeleter(void* p) {
+  // Some of our target environments would prefer a sized delete,
+  // but other target environments do not have that operator.
+  // Use an unsized delete until we get better agreement in the
+  // environments.
+  // See https://github.com/flutter/flutter/issues/100327
+  ::operator delete(p);
+}
+
+std::shared_ptr<DlPathEffect> DlPathEffect::From(SkPathEffect* sk_path_effect) {
+  if (sk_path_effect == nullptr) {
+    return nullptr;
+  }
+
+  SkPathEffect::DashInfo info;
+  if (SkPathEffect::DashType::kDash_DashType ==
+      sk_path_effect->asADash(&info)) {
+    auto dash_path_effect =
+        DlDashPathEffect::Make(info.fIntervals, info.fCount, info.fPhase);
+    info.fIntervals =
+        reinterpret_cast<DlDashPathEffect*>(dash_path_effect.get())
+            ->intervals();
```

Comment:
intervals() should be const (though it isn't currently - I mention that in another comment). So, you'll need a different, perhaps private API to get this pointer. Or just inline the "this+1" math here.